web/src/main/java/edu/softserve/zoo/controller/rest/LoginController.java

```diff
@@ -0,0 +1,62 @@
+package edu.softserve.zoo.controller.rest;
+
+import edu.softserve.zoo.dto.LoginRequestDto;
+import edu.softserve.zoo.dto.LoginResponseDto;
+import edu.softserve.zoo.web.security.AuthTokenUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import static edu.softserve.zoo.controller.rest.Routes.LOGIN;
+import static edu.softserve.zoo.controller.rest.Routes.USER;
+
+/**
+ * Controller for login requests.
+ * @author Ilya Doroshenko
+ */
+@RestController
+public class LoginController {
+
+    @Autowired
+    private AuthenticationManager authenticationManager;
+
+    @Autowired
+    private AuthTokenUtils authTokenUtils;
+
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    /**
+     * Handles user login requests. Generates token for remember-me functionality.
+     * @param loginRequestDto username and password
+     * @return authentication remember-me token
+     * @throws AuthenticationException if authentification fails
+     */
+    @RequestMapping(path = LOGIN, method = RequestMethod.POST)
+    public LoginResponseDto loginRequest(@RequestBody LoginRequestDto loginRequestDto) throws AuthenticationException {
```

Comment:
[See my comment on SSL usage](https://github.com/ITsvetkoFF/Kv-014/pull/133#discussion_r65511516)

As SSL effectively prevents MITM attack, HMAC (which with strong enough secret key, say random 128 bits, is bruteforce-proof on current tech level) ensures token integrity and authenticity - there is no need  to store token server-side

So from my point of view **logout** mechanism should be implemented client-side: just removing auth token from browser inner storage, no server changes (therefore requests) required.

**Login** request doesn't require authentication, so spring doesn't check if there is any token in headers, just returns new auth token.

Should it check if there is valid token and if so - deny request of a new one?